server/src/main/kotlin/org/itmodreamteam/myrest/server/repository/restaurant/ReservationRepository.kt

```diff
@@ -0,0 +1,28 @@
+package org.itmodreamteam.myrest.server.repository.restaurant
+
+import org.itmodreamteam.myrest.server.model.restaurant.Reservation
+import org.itmodreamteam.myrest.server.model.restaurant.Restaurant
+import org.itmodreamteam.myrest.server.model.restaurant.RestaurantTable
+import org.itmodreamteam.myrest.server.model.user.User
+import org.itmodreamteam.myrest.server.repository.JpaEntityRepository
+import org.itmodreamteam.myrest.shared.restaurant.ReservationStatus
+import org.springframework.data.jpa.repository.Query
+import org.springframework.data.repository.query.Param
+import java.time.LocalDateTime
+
+
+interface ReservationRepository : JpaEntityRepository<Reservation> {
+    // !(end1 <= start2 || end2 <= start1) = !(end1 <= start2) && !(end2 <= start1) = end1 > start2 && end2 > start1
+    @Query("select r from Reservation r where r.table = :table and r.status in :statuses and r.activeUntil > :activeFrom and :activeUntil > r.activeFrom")
+    fun findTableReservationsByStatusesAndTimeRangeOverlapping(
```

Comment:
Because the first one filters reservations by table (finds tables' reservations or reservation for table). If current naming is not clear maybe it is better to rename this method to something like `findReservationsForTableByStatusesAndTimeRangeOverlapping`

Reply:
I got it)